### PR TITLE
Add name argument in builder.play

### PR DESCRIPTION
**Pull request by peendebak**
### Summary

Add `name` argument to `qiskit.pulse.builder` the argument is passed to the underlying call to `instructions.Play`

### Details and comments

Default for the `name` argument is set to `None`, which is the same as for the `instructions.Play`

### CODE REVIEW

**Comment by taalexander** [comment]
Hello @peendebak, I was wondering if it would be possible to add this field for all similar instructions, eg., `set_phase`, `set_frequency`, `acquire`, etc.? In general, the `name` field is not often used for instructions outside of labeling metadata for visualization. The most important `name` is for the `Pulse` type for visualization.

**Comment by peendebak** [comment]
@taalexander I added the `name` argument to several more methods. The `acquire` I skipped because the name can already be passed via the `metadata`

I also skipped the various context manager methods, e.g.
```
@contextmanager
def frequency_offset(frequency: float, ...
```

File: qiskit/pulse/builder.py [hunk 1]
```diff
@@ -1599,13 +1604,14 @@ def shift_phase(phase: float,
         d0 = pulse.DriveChannel(0)
 
         with pulse.build() as pulse_prog:
-            pulse.shift_phase(math.pi, d0)
+            pulse.shift_phase(math.pi, d0, name='pi phase-shift')
```
Comment (taalexander) [inline qiskit/pulse/builder.py:1607]: I don't think we need to include this in the examples. In general, the `name` feature for instructions is not often used and the documentation mentions it.

File: test/python/pulse/test_builder.py [hunk 2]
```diff
@@ -367,6 +367,16 @@ def test_play_array_pulse(self):
 
         self.assertEqual(schedule, reference)
 
+    def test_play_name_argument(self):
```
Comment (taalexander) [inline test/python/pulse/test_builder.py:370]: Would you mind adding a quick test of `name` setting to every `test_<inst>` test? I would really appreciate this.

**Review by taalexander — CHANGES_REQUESTED** [review]
Almost there, just a couple small changes please and thank you 😄 .

**Review by taalexander — APPROVED** [review]
Looks great pending CI passing.

**Review by taalexander — APPROVED** [review]
Thank you very much for your contribution @eendebakpt, it is much appreciated 🥇 .

**Pull request merged**